Consumers/StripeMessageConsumer.cs

```diff
@@ -0,0 +1,180 @@
+                                LogLevels.Error
+                            );
+                            break;
+                    }
+                }
+                catch (Exception e)
+                {
+                    _logger.Log($"Error processing message from callbackQueue: {e}", LogLevels.Error);
+                }
+
+                await Task.Delay(2000);
+            }
+        }
+
+        async void InvoicePaymentSucceeded(Event stripeEvent, Message message)
+        {
+            var invoice = stripeEvent.Data.Object as Invoice;
+            var subscription = new SubscriptionModel
+            {
+                SubscriptionId = invoice!.SubscriptionId,
+                CustomerId = invoice.CustomerId,
+                Email = invoice.CustomerEmail,
+                UserStatus = invoice.Total == 0 ? UserStatus.FreeTrial : UserStatus.Subscribed,
+                InvoiceUrl = invoice.HostedInvoiceUrl,
+                StartDate = invoice.Lines.Data[0].Period.Start,
+                EndDate = invoice.Lines.Data[0].Period.End
+            };
+
+            await _subscriptionService.CreateSubscription(subscription);
+            await DeleteMessage(message.ReceiptHandle);
+        }
+
+        async void UpdateUserStatus(Event stripeEvent, Message message)
+        {
+            var updatedSubscription = stripeEvent.Data.Object as Subscription;
+            var response = await _subscriptionService.GetSubscriptionById(updatedSubscription!.Id);
+            var newUserStatus = UserStatus.Unknown;
+
+            var currentSubscription = response.Data;
+            if (currentSubscription == null) { return; }
+
+            // Determine new UserStatus based on the updated subscription's status
+            switch (updatedSubscription.Status) {
+                case "trialing":
+                    newUserStatus = UserStatus.FreeTrial;
+                    break;
+                case "paused":
+                    newUserStatus = UserStatus.FreeTrialEnded;
+                    break;
+                case "active":
+                    newUserStatus = UserStatus.Subscribed;
+                    break;
+                case "canceled":
+                    newUserStatus = currentSubscription.UserStatus == UserStatus.FreeTrial ? UserStatus.FreeTrialEnded : UserStatus.SubscriptionEnded;
+                    break;
+                case "unpaid":
+                    newUserStatus = UserStatus.NotPaid;
+                    break;
+                case "past_due":
+                    newUserStatus = UserStatus.NotPaid;
+                    break;
+                default:
+                    newUserStatus = UserStatus.Unknown;
+                    break;
+            }
+
+            // Create recurring job if UserStatus moves to FreeTrial or Subscribed
+            if (newUserStatus == UserStatus.FreeTrial || newUserStatus == UserStatus.Subscribed)
```

Comment:
I'm not sure this is necessary since we are only considering the updated (new) user status. as long as the new user status shows them in the free trial or as subscribed we will register their job